lib/services/static.js

```diff
@@ -11,7 +11,109 @@ var invariant = require('../../vendor/invariant'),
 var MapboxStatic = makeService('MapboxStatic');
 
 /**
- * Determine a URL for a static map image, using the [Mapbox Static Map API](https://www.mapbox.com/developers/api/static/).
+ * Determine a URL for a static map image, using the [Mapbox Static Map API](https://www.mapbox.com/api-documentation/#static).
+ *
+ * @param {string} username
+ * @param {string} styleid
+ * @param {number} width width of the image
+ * @param {number} height height of the image
+ *
+ * @param {Object|string} position either an object with longitude and latitude members, or the string 'auto'
+ * @param {number} position.longitude east, west bearing
+ * @param {number} position.latitude north, south bearing
+ * @param {number} position.zoom zoom level
+ * @param {number} position.bearing
+ * @param {number} position.pitch
+ *
+ * @param {Object} options all map options
+ * @param {boolean} [options.retina=false] whether to double image pixel density
+ *
+ * @param {Array<Object>} [options.markers=[]] an array of simple marker objects as an overlay
+ * @param {Object} [options.geojson={}] geojson data for the overlay
+ * @param {Object} [options.path={}] a path and
+ * @param {Array<Object>} options.path.geojson data for the path as an array of longitude, latitude objects
+ * @param {Array<Object>} options.path.style optional style definitions for a path
+ * @param {boolean} options.attribution controlling whether there is attribution on the image; defaults to true
+ * @param {boolean} options.logo controlling whether there is a Mapbox logo on the image; defaults to true
+ * @param {string} options.before_layer value for controlling where the overlay is inserted in the style
+ *
+ * @returns {string} static map url
+ * @memberof MapboxClient
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ */
+MapboxStatic.prototype.getStaticURL = function(username, styleid, width, height, position, options) {
+  invariant(typeof username === 'string', 'username option required and must be a string');
+  invariant(typeof styleid === 'string', 'styleid option required and must be a string');
+  invariant(typeof width === 'number', 'width option required and must be a number');
+  invariant(typeof height === 'number', 'height option required and must be a number');
+
+  var defaults = {
+    retina: ''
+  };
+
+  var xyzbp;
+
+  if (position === 'auto') {
+    xyzbp = 'auto';
+  } else {
+    invariantLocation(position);
+    xyzbp = position.longitude + ',' + position.latitude + ',' + position.zoom
+          + (('bearing' in position) ? ',' + position.bearing : '')
```

Comment:
fixed together with a change to deal with being passed bearing / pitch as exclusive options.